ehcache/src/play/cache/EhCacheImpl.java

```diff
@@ -62,13 +64,17 @@ private EhCacheImpl() {
     this.cache = cacheManager.getCache(cacheName, String.class, ValueWrapper.class);
   }
 
-  public static EhCacheImpl getInstance() {
+  public static EhCacheImpl instance(@SuppressWarnings("unused") Properties playProperties) {
+    if (uniqueInstance == null) {
+      uniqueInstance = new EhCacheImpl();
+    }
     return uniqueInstance;
   }
 
-  public static EhCacheImpl newInstance() {
-    uniqueInstance = new EhCacheImpl();
-    return uniqueInstance;
+  /** Should only be used in tests */
+  @VisibleForTesting
+  public static EhCacheImpl testInstance() {
+    return new EhCacheImpl();
   }
```

Comment:
@cies Why method `testInstance()` is needed? It's simple to just call `new EhCacheImpl()` instead of `EhCacheImpl.testInstance()`.

Reply:
@cies I would make method `testInstance()` _package-private_ instead of public (it means: without "public", "protected" or "private" modifier). Then it's visible from unit-tests, and at least not visible from other packages.

Reply:
Nice one. Fixed.